src/ripple/overlay/impl/PeerImp.h

```diff
@@ -198,6 +199,26 @@ class PeerImp
 
     friend class OverlayImpl;
 
+    class Metrics {
+    public:
```

Comment:
I recommend adding:
```
        Metrics() = default;
        Metrics(Metrics const&) = delete;
        Metrics& operator=(Metrics const&) = delete;
```

This helps reader know that this class will not be copied/moved.